MetaMask/metamask-extension
# feat: Bump `queued-request-controller` to `^1.0.0`

## **Description**

Bumps `@metamask/queued-request-controller` to `^1.0.0` in order to handle the case where `eth_requestAccounts` requests that would not cause a confirmation to be presented to the user (because they have permitted some accounts for that dapp) would be stuck behind the request queue when they could just be passed through.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25310?quickstart=1)

## **Related issues**

See: https://github.com/MetaMask/core/pull/4423
Fixes: https://github.com/MetaMask/metamask-extension/issues/25407

## **Manual testing steps**

1. Open https://metamask.github.io/test-dapp/ and https://app.odos.xyz/, connect to both
1. On the Test Dapp, send an EIP 1559 txn, but do not confirm/reject the confirmation nor close the popup
1. Reload ODOS
1. ODOS should show that is connected to MetaMask in the top right of the webpage
1. Open console on the ODOS page
1. run `await window.ethereum.request({method: 'eth_requestAccounts'})`, it should return immediately with the permitted accounts


## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/MetaMask/metamask-extension/assets/918701/b76607ef-b0a3-4943-b1d5-85f8cd766343




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
